### PR TITLE
Allow generation into sub-namespaces

### LINKED ISSUE #2: Support for configurable Flex directories
Currently, the generators are hardcoding the `src` folder for the source. But `SRC_DIR` is configurable in Flex.
The bundle should read the configured value in the composer.json instead, to account for Flex configuration.

### PR DESCRIPTION
**Pull request by weaverryan**
Three changes:

1) Generation is now allowed into sub-namespaces: e.g. `make:controller` passing it `Admin\MainController` would generate into `src/Controller/Admin/MainController.php`.

2) You can pass absolute class names now too - e.g. `OtherNamespace\Controller\FooController`. Maker reads the Composer autoloader to figure out where this should live - e.g. `lib/Controller/FooController.php`. This fixes #2: Maker follows your autoloading rules to find *where* to generate things.

3) [BC] Break for anyone with custom makes: the `MakerInterface` was updated to be much simpler. Instead of `getParameters()`, `getFiles()` and `writeSuccessMessage()`, you just implement `generate()` and do it all in one method. I think using this library to create custom maker commands is still quite uncommon (we are listed as dependencies of VERY few projects on Packagist), so I think we should make this change versus keeping BC code paths for a long time in the future.

Cheers!

### CODE REVIEW

**Comment by iSDP** [comment]
<3

**Comment by mjemajr** [comment]
Hi @weaverryan, it seems that make:controller Admin\MainController generates the controller into the src/Controller/AdminMainController.php namespace instead of src/Controller/Admin. To have it inside the Admin sub-namespace i had to use double backslashes. Is this the default behavior?

**Comment by weaverryan** [comment]
Yes, the double back slash thing is just because that’s how arguments work on the terminal - you need two \ to be seen as one. Nothing we can do about that :/.

However, if you just run make:controller and then answer the question (instead of passing as an arg), I believe it works with a single \

File: src/FileManager.php [hunk 1]
```diff
@@ -20,17 +21,19 @@
...
     /** @var SymfonyStyle */
     private $io;
 
+    private static $classLoader;
+
     public function __construct(Filesystem $fs, string $rootDirectory)
     {
         $this->fs = $fs;
-        $this->rootDirectory = $rootDirectory;
+        $this->rootDirectory = rtrim($this->realpath($rootDirectory).'/');
```
[the remark this answers is not in the document]
Comment (weaverryan) [reply src/FileManager.php]: Hmm, looks like you might be right. Could you open a PR?

**Comment by mjemajr** [comment]
Thank you very much, answering the question works perfectly.

**Comment by SeanHolt** [comment]
> However, if you just run make:controller and then answer the question (instead of passing as an arg), I believe it works with a single \

Just for clarity, this works precisely as described.
Thanks for everybody's hard work on this project!

**Pull request merged**